opm/simulators/timestepping/TimeStepControl.cpp

```diff
@@ -206,7 +206,7 @@ namespace Opm
             const double kD = 0.01 ;
             const double newDt = (dt * std::pow( errors_[ 1 ] / errors_[ 2 ], kP ) *
                                  std::pow( tol_         / errors_[ 2 ], kI ) *
-                                 std::pow( errors_[0]*errors_[0]/errors_[ 1 ]/errors_[ 2 ], kD ));
+                                 std::pow( errors_[1]*errors_[1]/errors_[ 0 ]/errors_[ 2 ], kD ));
```

Comment:
I've looked a little bit closer at this.  Suppose `errors_[0] == 0`, which it theoretically could be since the conditional above only checks [`errors_[1]` and `errors_[2]`](https://github.com/OPM/opm-simulators/blob/c1604161fe3a845280121c7e19235247ceaa05b1/opm/simulators/timestepping/TimeStepControl.cpp#L195).  Then `newDt` will be non-finite, typically `inf`.  Is that reasonable here?

Reply:
I guess the errors_[1] and errors[2] were included in the conditional because they appeared in the denominators of the PID formula, and hence it was only relevant to check those. However, having changed indexing, errors_[0] (and not errors_[1]) appears in the denominator. It might therefore be reasonable to add errors[0] to (and possibly just remove errors_[1] from) the conditional.

Reply:
> It might therefore be reasonable to add `errors[0]` to [...] the conditional.

Yes, that's probably needed.

> and possibly just remove `errors_[1]` from the conditional

I'd be wary of doing that.  If `errors_[1] = 0` we'll get `newDt = 0` as well and that seems unexpected to me.  Maybe I'm just misunderstanding something.

Reply:
You're probably right. It appears that that might have been a problem earlier, though. Regardless, it certainly shouldn't hurt to add `errors_[0] == 0` to the conditional.

I added it now.

Reply:
> It appears that that (`errors_[0] = 0` giving `newDt = 0`) might have been a problem earlier, though

Agreed.  I only remarked on this possibility because we're in the area now so it's an opportunity to fix it.  For what it's worth, I tried this PR on a field case and ran into that exact problem.  Otherwise I might not have noticed it.